hoot-core/src/main/cpp/hoot/core/conflate/DiffConflator.cpp

```diff
@@ -283,13 +266,16 @@ Change DiffConflator::_getChange(ConstElementPtr pOldElement,
   // This may seem a little weird, but we want something very specific here.
   // We want the old element as it was... with new tags.
 
-  // Copy the old one
+  // Copy the old one to get the geometry
   ElementPtr pChangeElement (pOldElement->clone());
 
   assert(pChangeElement->getId() == pOldElement->getId());
 
-  // Overwrite all old tags
-  pChangeElement->setTags(pNewElement->getTags());
+  // Need to merge tags into the new element
+  // Keeps all names, chooses tags1 in event of a conflict.
+  Tags newTags = TagComparator::getInstance().overwriteMerge(pNewElement->getTags(),
+                                                             pOldElement->getTags());
+  pChangeElement->setTags(newTags);
```

Comment:
This is great.  I think this is the right way to merge the tags to get the correct diff.